src/views/Affiliates/AffiliatesLeaderboard.tsx

```diff
@@ -0,0 +1,320 @@
+
+                {accountStats?.account && account === accountStats.account && (
+                  <Tag className="bg-color-accent">{stringGetter({ key: STRING_KEYS.YOU })}</Tag>
+                )}
+              </TableCell>
+            );
+          },
+          allowsSorting: false,
+        },
+        {
+          columnKey: 'account',
+          label: stringGetter({ key: STRING_KEYS.ACCOUNT }),
+          allowsSorting: false,
+
+          renderCell: ({ code }) => <$AccountOutput type={OutputType.Text} value={code} />,
+        },
+        {
+          columnKey: 'total-earnings',
+          label: stringGetter({ key: STRING_KEYS.TOTAL }),
+          allowsSorting: false,
+
+          renderCell: ({ totalEarnings, totalReferredUsers }) => (
+            <TableCell className="flex flex-col justify-start text-end">
+              <div className="w-full">
+                <$EarningsOutput
+                  type={OutputType.CompactFiat}
+                  value={totalEarnings}
+                  slotRight={
+                    <span className="ml-0.25 text-color-text-2">
+                      {stringGetter({ key: STRING_KEYS.EARNINGS }).toLocaleLowerCase()}
+                    </span>
+                  }
+                />
+              </div>
+              <div className="w-full text-color-text-0">
+                {totalReferredUsers.toLocaleString()}{' '}
+                {stringGetter({ key: STRING_KEYS.USERS_REFERRED })}
+              </div>
+            </TableCell>
+          ),
+        },
+      ]
+    : [
+        {
+          columnKey: 'rank',
+          label: stringGetter({ key: STRING_KEYS.RANK }),
+          allowsSorting: false,
+          renderCell: ({ rank, account }) => (
+            <TableCell className="align-center flex text-color-text-1 font-base-medium">
+              {rank}
+              {accountStats?.account && account === accountStats.account && (
+                <Tag className="bg-color-accent">{stringGetter({ key: STRING_KEYS.YOU })}</Tag>
+              )}
+            </TableCell>
+          ),
+        },
+        {
+          columnKey: 'account',
+          label: stringGetter({ key: STRING_KEYS.ACCOUNT }),
+          allowsSorting: false,
+
+          renderCell: ({ code }) => <$AccountOutput type={OutputType.Text} value={code} />,
+        },
+        {
+          columnKey: 'total-earnings',
+          label: stringGetter({ key: STRING_KEYS.TOTAL_EARNINGS }),
+          allowsSorting: false,
+
+          renderCell: ({ totalEarnings }) => (
+            <$EarningsOutput type={OutputType.CompactFiat} value={totalEarnings} />
+          ),
+        },
+        {
+          columnKey: 'ref-vol',
+          label: stringGetter({ key: STRING_KEYS.VOLUME_REFERRED }),
+          allowsSorting: false,
+
+          renderCell: ({ referredVolume }) => (
+            <$NumberOutput type={OutputType.CompactFiat} value={referredVolume} />
+          ),
+        } as ColumnDef<IAffiliateStats>,
```

Comment:
would `satisfies` work instead so we don't have to typecast?

Reply:
YES! No more casts!

Reply:
Should we add `satisfies` to the entire array instead of individual objects?